algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -63,8 +61,26 @@ object QTree {
   def apply(k: Double): QTree[Double] = apply(k -> k)
 }
 
-class QTreeSemigroup[A: Monoid](k: Int) extends Semigroup[QTree[A]] {
+class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extends Semigroup[QTree[A]] {
```

Comment:
It would wind up being just longer and redundant, you'd need to add another val to the class or def that was just 
val underlyingMonoid = implicitly[Monoid[A]] 

In either case it'll be picked implicitly, indeed the A: Monoid is just compiler magic for inserting it into the implicit block of the args, but with effectively an anonymous name.